proxy/proxy.go

```diff
@@ -30,6 +35,17 @@ func MiddlewareSetUpstream(upstream *url.URL) Middleware {
 	}
 }
 
+func MiddlewareSetPrefixHeader(headerKey string) Middleware {
```

Comment:
`headerKey` is a bit confusing as it seems to mix `header` and "context key". Let's be consistent with usage elsewhere and call it `header`
```suggestion
func MiddlewareSetPrefixHeader(header string) Middleware {
```